DirectProgramming/DPC++FPGA/Tutorials/DesignPatterns/zero_copy_data_transfer/src/CMakeLists.txt

```
@@ -20,16 +20,16 @@ endif()
 # this tutorial requires USM host allocations. Check the BSP name (which should contain the text 'usm')
 # to ensure the BSP has the required support. Allow the user to define USM_HOST_ALLOCATIONS_ENABLED
 # to override this check (e.g., cmake .. -DUSM_HOST_ALLOCATIONS_ENABLED=1)
-if(NOT FPGA_BOARD MATCHES ".usm.*" AND NOT DEFINED USM_HOST_ALLOCATIONS_ENABLED)
+if(NOT FPGA_BOARD MATCHES ".usm.*" AND (NOT DEFINED USM_HOST_ALLOCATIONS_ENABLED OR USM_HOST_ALLOCATIONS STREQUAL "0"))
```

Looking more closely, I'm not sure why you introduced the USM_HOST_ALLOCATIONS variable, as distinct from USM_HOST_ALLOCATIONS_ENABLED?  I recommend removing all references to USM_HOST_ALLOCATIONS (including those below), only using USM_HOST_ALLOCATIONS_ENABLED, and changing this line of code as follows:
```suggestion
if((NOT FPGA_BOARD MATCHES ".usm.*") AND (NOT USM_HOST_ALLOCATIONS_ENABLED))
```

---

That was the intention, but I messed it up. I think I have done so now.